decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/utils/Barcodes.java

```diff
@@ -0,0 +1,106 @@
+package ca.on.oicr.pde.deciders.utils;
+
+import ca.on.oicr.pde.deciders.data.Barcode;
+import ca.on.oicr.pde.deciders.data.BasesMask;
+import ca.on.oicr.pde.deciders.exceptions.DataMismatchException;
+import com.google.common.base.Strings;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ *
+ * @author mlaszloffy
+ */
+public class Barcodes {
+
+    public static Barcode applyBasesMask(Barcode barcode, BasesMask basesMask) throws DataMismatchException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)